datahub-classify/src/datahub_classify/infotype_helper.py

```diff
@@ -536,3 +544,440 @@ def inspect_for_age(metadata, values, config):  # noqa: C901
 
     return confidence_level, debug_info
+
+
+def inspect_for_iban(metadata, values, config):  # noqa: C901
+    prediction_factors_weights = config[PREDICTION_FACTORS_AND_WEIGHTS]
+    debug_info: Dict[str, Any] = {}
+
+    # Value Logic
+    if prediction_factors_weights.get(VALUES, 0) > 0:
+        values_score = 0.0
+        try:
+            if config[VALUES][PREDICTION_TYPE] == "regex":
+                values_score = match_regex_for_values(values, config[VALUES][REGEX])
+            elif config[VALUES][PREDICTION_TYPE] == "library":
+                iban_score = 0.0
+                for val in values:
+                    try:
+                        if IBAN(val, allow_invalid=True).is_valid:
+                            iban_score += 1.0
+                    except Exception:
+                        pass
+                values_score = iban_score / len(values)
+            else:
+                raise Exception(
+                    "Inappropriate values_prediction_type %s"
+                    % config[VALUES][PREDICTION_TYPE]
+                )
+        except Exception as e:
+            logger.error(f"Column {metadata.name} failed due to {e}")
+        values_score = np.round(values_score, 2)
+        debug_info[VALUES] = values_score
+
+    # Name Logic
+    if prediction_factors_weights.get(NAME, 0) > 0:
+        if not metadata.name or not metadata.name.strip():
+            debug_info[NAME] = f"0.0 (Blank {NAME} Metadata)"
+        else:
+            debug_info[NAME] = match_regex(metadata.name, config[NAME][REGEX])
+
+    # Description_Logic
+    if prediction_factors_weights.get(DESCRIPTION, 0) > 0:
+        if not metadata.description or not metadata.description.strip():
+            debug_info[DESCRIPTION] = f"0.0 (Blank {DESCRIPTION} Metadata)"
+        else:
+            debug_info[DESCRIPTION] = match_regex(
+                metadata.description, config[DESCRIPTION][REGEX]
+            )
+
+    # Datatype_Logic
+    if prediction_factors_weights.get(DATATYPE, 0) > 0:
+        if not metadata.datatype or not metadata.datatype.strip():
+            debug_info[DATATYPE] = f"0.0 (Blank {DATATYPE} Metadata)"
+        else:
+            debug_info[DATATYPE] = match_datatype(
+                metadata.datatype, config[DATATYPE][TYPE]
+            )
+
+    confidence_level = 0
+    for key in debug_info.keys():
+        if type(debug_info[key]) != str:
+            confidence_level += prediction_factors_weights[key] * debug_info[key]
+    confidence_level = np.round(confidence_level, 2)
+
+    return confidence_level, debug_info
+
+
+def inspect_for_vehicle_identification_number(metadata, values, config):  # noqa: C901
+    prediction_factors_weights = config[PREDICTION_FACTORS_AND_WEIGHTS]
+    debug_info: Dict[str, Any] = {}
+
+    # Value Logic
+    if prediction_factors_weights.get(VALUES, 0) > 0:
+        values_score = 0.0
+        try:
+            if config[VALUES][PREDICTION_TYPE] == "regex":
```

Comment:
this code seems highly repetitive with tons of copy-paste-modify - consider refactoring

Reply:
We are considering this in our improvements TODOs and will address this in the next PR

Reply:
sure thing - please just add some TODOs in the code that say that

Reply:
Added a TODO in the infotype_helper.py